### PR TITLE
GH-434 check binary existence in filestore

### PR DESCRIPTION
**Pull request by danielmkn**
added `check_binary_existence_in_filestore` field to pull and push replications. Updated documentation, tests modified.
Added deprecation note to .md documents for old replication resources.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
+## 6.7.2 (May 13, 2022). Tested on Artifactory 7.38.8
+
+IMPROVEMENTS:
+
+* resource/artifactory_pull_replication.go and resource/artifactory_push_replication.go: Add new attribute `check_binary_existence_in_filestore`. 
+PR: [#459](https://github.com/jfrog/terraform-provider-artifactory/pull/459). 
+Issue [#434](https://github.com/jfrog/terraform-provider-artifactory/issues/434)
+
+## 6.7.1 (May 13, 2022). Tested on Artifactory 7.38.8
```
Comment (alexhung) [inline CHANGELOG.md:9]: Why is this change in this PR? You may need to refresh your branch

File: pkg/artifactory/resource/replication/resource_artifactory_pull_replication.go [hunk 2]
```diff
@@ -71,6 +71,13 @@ var pullReplicationSchema = map[string]*schema.Schema{
 		Optional:    true,
 		Description: "Proxy key from Artifactory Proxies setting",
 	},
+	"check_binary_existence_in_filestore": {
+		Type:     schema.TypeBool,
+		Optional: true,
+		Default:  false,
```
Comment (alexhung) [inline pkg/artifactory/resource/replication/resource_artifactory_pull_replication.go:77]: Values are not aligned :)

File: pkg/artifactory/resource/replication/resource_artifactory_push_replication.go [hunk 3]
```diff
@@ -136,6 +137,12 @@ var pushReplicationSchema = map[string]*schema.Schema{
 		Optional:    true,
 		Description: "Proxy key from Artifactory Proxies setting",
 	},
+	"check_binary_existence_in_filestore": {
+		Type:     schema.TypeBool,
+		Optional: true,
+		Description: "When true, enables distributed checksum storage. For more information, see " +
```
Comment (alexhung) [inline pkg/artifactory/resource/replication/resource_artifactory_push_replication.go:143]: Align the values 😄

File: pkg/artifactory/resource/repository/federated/resource_artifactory_federated_generic_repository.go [hunk 4]
```diff
@@ -11,8 +11,9 @@ import (
 )
 
 func ResourceArtifactoryFederatedGenericRepository(repoType string) *schema.Resource {
+	localRepoSchema := local.GetSchemaByRepoType(repoType)
 
-	var federatedSchema = util.MergeSchema(local.BaseLocalRepoSchema, map[string]*schema.Schema{
+	var federatedSchema = util.MergeSchema(localRepoSchema, map[string]*schema.Schema{
```
Comment (alexhung) [inline pkg/artifactory/resource/repository/federated/resource_artifactory_federated_generic_repository.go:16]: This is part of an existing commit. This means your branch is not freshed.

**Review by alexhung — CHANGES_REQUESTED** [review]
Your changes looks fine with very minor updates needed.

You need to refresh your branch so it doesn't contain changes from the last PRs in `master`

**Comment by danielmkn** [comment]
I'll open a new PR from a different branch.

**Pull request closed without merging**